### PR TITLE
Week 3: Presentation Proposal

### PR DESCRIPTION
**Pull request by einbergisak**
# Assignment Proposal

## Title

Exploring Canary Deployments with Flagger

## Names and KTH ID

- Johannes Matsson (jmatsso@kth.se)
- Isak Einberg (ieinberg@kth.se)

## Deadline

- Week 3

## Category

- Presentation

## Description

In this presentation we will explore **Canary Deployments**, focusing on how they function, as well as their key advantages. The presentation will include a comparison to the alternative blue-green deployment strategy. 

We aim to present how Canary Deployments can be used as a safe and efficient strategy for introducing changes in a production environment, preferrably using a real world example.

To summarize, we aim to cover:
- The mechanics of Canary Deployments.
- How they minimize risks during deployment.
- Example of use cases (preferrably a real-world scenario).
- A comparison with blue-green deployment.

**Relevance**

Canary Deployments are very relevant, as they allow teams to gradually introduce new changes to a subset of users, thus **reducing the risk of widespread failures**. It is therefore highly relevant for organizations aiming to achieve **continuous delivery**, by ensuring safer rollouts.

### CODE REVIEW

**Comment by algomaster99** [comment]
Hi! Any specific technology that you plan to explore to do canary deployment?

**Comment by einbergisak** [comment]
@algomaster99 Hi! We plan on presenting an example with Flagger and Istio. Do you think it would be better if we focus the presentation on a specific technology (like Flagger), instead of prioritizing explaining Canary deployment itself?

Thanks!

**Comment by algomaster99** [comment]
@einbergisak I asked about technology because there is [another task](https://github.com/KTH/devops-course/pull/2414) about canary deployment and I did not want the technologies to overlap. :sweat_smile: 

> Do you think it would be better if we focus the presentation on a specific technology (like Flagger), instead of prioritizing explaining Canary deployment itself?

Since this is a technical talk, you explain the concepts wrt a technology. Briefing audience about canary deployment is good but we don't see how exactly it is beneficial. I will merge once you understand my comment :)

**Comment by einbergisak** [comment]
@algomaster99 We understand! We'll focus on Canary Deployment for Flagger specifically.

**Comment by einbergisak** [comment]
Oh an one other thing, will we have time to present on Wednesday this week? I saw that there are already 20+ people with merged PRs referring to Week 3

**Comment by algomaster99** [comment]
The presentations that remain are presented in the lab next day :) @Deee92 , correct me if I am wrong.
